### ISSUE TITLE
Issue in LW optimal shrinkage

### ISSUE THREAD

**Issue by tlienart** [mateuszbaran/CovarianceEstimation.jl#12]
I'm in the process of trying to clarify the shrinkage parameter and I believe there's a mistake in the computation of ϑhatjj. It was:

```julia
ϑhatii  = mean([(Xint[i,t]^2 - C[i,i])*(Xint[i,t]*Xint[j,t] - C[i,j]) for i in 1:N, j in 1:N] for t in 1:Tnum)
ϑhatjj = mean([(Xint[i,t]^2 - C[j,j])*(Xint[i,t]*Xint[j,t] - C[i,j]) for i in 1:N, j in 1:N] for t in 1:Tnum)
```

on the second line it should be `Xint[j, t]`.

Screenshot of the pdf:

![screen shot 2018-12-09 at 1 26 44 pm](https://user-images.githubusercontent.com/10897531/49692749-2ea53a00-fbb6-11e8-9752-2db64b7731e4.png)

I'm writing a PR (with some other fixes too), opening this issue for the records, should be closed soon.

**Comment by tlienart** [comment]
Actually I think there's an issue in the computation of `ρhatpart2`, the paper skips `j=i` in the inner sum but this is not done in the code.

**Comment by tlienart** [comment]
I also disagree with returning `0.0` if `γ̂` is very small, indeed if `π̂ - ρ̂` is negative (which can happen) then you'd clamp at 0, whereas if it's positive then you'd clamp at 1.

**Comment by mateuszbaran** [comment]
Thank you, well spotted. In case of returning 0 when `γhat` is very small, it doesn't really matter when it's not returned to the user. In this case both covariance matrix and target matrix are very similar (since `γhat` is a squared norm of their difference).

By the way, do you know of an implementation that fully conforms to the paper? I read for example the documentation for sklearn's implementation and they use a different shrinkage target. There is also an implementation in R but it's GPL so I'm not really comfortable reading their code to see if it actually conforms.

**Comment by tlienart** [comment]
this is in #15 and can be closed upon merging.

Re `γ̂` see what I've done in the PR and let me know what you think, my issue is that it could be either total shrinkage (likely not but possible) or no shrinkage (the case you assume).

**Comment by mateuszbaran** [comment]
I've seen it and with `lw_optimalshrinkage` as a separate function your solution makes more sense than mine.

**Comment by mateuszbaran** [comment]
Fixed by PR #15
